desk/app/channels.hoon

```diff
@@ -617,26 +506,31 @@
       ca-abet:(ca-join:ca-core [nest group.a-channel]:a-channels)
     ::  upconvert emoji
     ::
-    =?  a-channel.a-channels  ?=([%post %add-react *] a-channel.a-channels)
-      %_  a-channel.a-channels
-        q.c-post
+    =/  new-a-channels=a-channels:c
+      a-channels
+    ?>  ?=([%channel *] new-a-channels)
+    =?  new-a-channels  ?=([%post %add-react *] a-channel.a-channels)
+      ?>  ?=([%post %add-react *] a-channel.new-a-channels)
+      %_  new-a-channels
```

Comment:
Not sure I understand why you do the `=/  new-a-channels` here. In this new/current code, we never use `a-channels` again, so it's not like we're keeping a copy of the original data around for reference, right?

The code from before your refactor commit, where it just modified the `a-channels` in-place, was fine.

Reply:
This has actually been rewritten: the current version does not look like that anymore (I am looking at `bb24466`). Can you please take a look again?